src/Employer/Employer.Web/Controllers/VacancyPreviewController.cs

```diff
@@ -55,6 +55,8 @@ public async Task<IActionResult> Submit(SubmitEditModel m)
 
                 if (response.Data.HasLegalEntityAgreement == false)
                     return RedirectToRoute(RouteNames.LegalEntityAgreement_HardStop_Get);
+
+                throw new Exception("Unknown submit state");
```

Comment:
We have an `InvalidStateException` class

Reply:
An `InvalidStateException`  gets turned into a dashboard message saying a "colleague has made changes..." This exception should never fire.